client/index.js

```diff
@@ -0,0 +1,70 @@
+const axios = require("axios");
+
+async function getKey(request, token, key) {
+  let query = token ? `?token=${token}` : "";
+  const response = await request.get(`/keys/${key}${query}`);
+  const result = response.data.value.value;
+  return result;
+}
+
+async function putKey(request, token, key, value) {
+  let query = token ? `?token=${token}` : "";
+  await request.put(`/keys/${key}${query}`, { value });
+  return;
+}
+
+class Connection {
+  constructor(request) {
+    this.request = request;
+  }
+
+  async get(key) {
+    return getKey(this.request, undefined, key);
+  }
+
+  async writeTransaction() {
+    const response = await this.request.post("/begin-write-transaction");
+    const token = response.data;
+    return new WriteTransaction(this.request, token);
+  }
```

Comment:
Why all the inheritance @davazp ? It's a hack project. I don't much care. Still, this function used to be three lines of code, now it's three levels of classes. If you'd like to relate the transactions to a connection, how about we pass the connection as an argument?